src/lib.rs

```diff
@@ -282,6 +282,10 @@
 //! it will only happen in a minor or major version release.
 
 #![warn(missing_docs)]
+// `missing_crate_level_docs` was renamed with a `rustdoc::` prefix in rustc 1.55, but isn't
+// supported in the MSRV.
+// TODO: remove this if/when the MSRV is 1.55+.
+#![allow(renamed_and_removed_lints)]  
```

Comment:
I think one possible way to avoid this would be to use `#![cfg_attr(docsrs, warn(rustdoc::missing_crate_level_docs))]`, and then to add `--cfg docsrs` to the end of the invocations in `check-rustdoc.sh`. We'd also need to use `+nightly` in that script too once we do that, since we use some nightly only stuff for our docs generation. This would be a general improvement anyways I think, since it would bring our docs linting closer to what docs.rs actually does to generate the documentation and also [how we lint in the BSON library](https://github.com/mongodb/bson-rust/blob/master/.evergreen/check-rustdoc.sh#L6).

Reply:
Ah, that is nicer, thank you - done.